jwst/cube_build/blot_cube_build.py

```diff
@@ -309,34 +279,49 @@ def blot_images_nirspec(self):
 
                 # median cube ra,dec,wave -> x_slice, y_slice
                 x_slice, y_slice = slice_wcs.invert(ra_use, dec_use, wave_use)
-
                 x_slice = np.ndarray.flatten(x_slice)
                 y_slice = np.ndarray.flatten(y_slice)
                 flux_slice = np.ndarray.flatten(flux_use)
 
+                # only use values what fall in bounding box of the slice
                 xlimit, ylimit = slice_wcs.bounding_box
                 xuse = np.logical_and(x_slice >= xlimit[0], x_slice <= xlimit[1])
                 yuse = np.logical_and(y_slice >= ylimit[0], y_slice <= ylimit[1])
+                fgood = (flux_slice != 0)  # edge cube pixel may = 0
 
-                fuse = np.logical_and(xuse, yuse)
+                # fuse = np.logical_and(xuse, yuse)
+                fuse = np.where(fgood & xuse & yuse)
                 x_slice = x_slice[fuse]
                 y_slice = y_slice[fuse]
                 flux_slice = flux_slice[fuse]
 
-                xstart = 0
-                nn = flux_slice.size
-                for ipt in range(nn):
-                    blot_cube.blot_overlap(ipt, xstart,
-                                           xcenter, ycenter,
-                                           x_slice, y_slice,
-                                           flux_slice,
-                                           blot_xsize,
-                                           blot_flux, blot_weight)
-                ts1 = time.time()
-                log.debug(f"Time to blot 1 slice on NIRspec  =  {ts1-ts0:.1f}")
+                nslice = len(x_slice)
+                for i in range(nslice):
+                    x_total.append(x_slice[i])
+                    y_total.append(y_slice[i])
+                    flux_total.append(flux_slice[i])
```

Comment:
As explained in https://github.com/spacetelescope/jwst/pull/6253#discussion_r679538474, this can be rewritten more compactly as 
```suggestion
                x_total.extend(x_slice)
                y_total.extend(y_slice)
                flux_total.extend(flux_slice)
```

Reply:
I originally had what you wrote but then I got a list of arrays rather than 1 long array and it failed in the c code. How do I convert a list of arrays to a single  continuous list or a single continuous array ? I tried  to 'flatten' it but that did not work either.